rock_paper_scissors/api/models.py

```diff
@@ -0,0 +1,65 @@
+"""
+Relationships with Foreign Keys, esp. relationship() and FK values
+Why do we need to establish columns for players in game?
+difference with many to one and one to many -- if you back_populate either relationship type,
+    does it matter which you use?
+
+"""
+
+from sqlalchemy import Column, Integer, String
+from sqlalchemy import Enum
+from sqlalchemy import ForeignKey
+from sqlalchemy.ext.declarative import as_declarative
+from sqlalchemy.orm import relationship
+
+
+@as_declarative()
+class ModelBase(object):
+    def __tablename__(self):
+        return self.__class__.__name__.lower()
+
+
+class Game(ModelBase):
+    id = Column(Integer, primary_key=True)
+    player_1_id = Column(Integer, ForeignKey('user.id'))
+    player_2_id = Column(Integer, ForeignKey('user.id'))
+    round_id = Column(Integer, ForeignKey('round.id'))
+    rounds_choice = Column(Integer)
+    rounds_decrement = Column(Integer)
+    current_round = Column(Integer)
+    winner = Column(String, ForeignKey('user.id'))
+
+    round = relationship('Round', back_populates='game')
+    user = relationship('User', back_populates='game', uselist=False)
```

Comment:
1. To be honest, I _don't_ really think about it that much, at least not as an up-front design concept. To be clear, these terms describe a particular directional relationship- if `Game` has many `Rounds`, then "Game to Round" is a "One to Many" relationship. Conversely, "Round to Game" is a "Many to One" relationship. It's worth also noting that there are more complex relationship scenarios that don't really fit any of these terms.
2. It's important to remember that foreign keys exist within the database and essentially mean that "any value in this FK column actually is representative of a value that lives in a different column (of another table)". Most (all?) relational databases enforce that if you have a `dog.breed_id` column that is a foreign key to `breed.id`, the database will reject any value added to `dog.breed_id` unless that value also exists in `breed.id`. 
So if I have this `dog` table:

| id | name   | breed_id |
|----|-----------|-------------|
| 1  | Fido     | 3             |
| 2  | Max     | 1             |
| 3  | Rover  | 3             |
| 4  | Lassie | 2             |

And this `breed` table:

| id | type              |
|----|------------------|
| 1  | poodle          |
| 2  | border collie |
| 3  | dalmatian     |

The database will prevent me from adding a row like this to the `dog` table:

| id | name        | breed_id |
|----|---------------|--------------|
| 5 | Rin Tin Tin | 10           |

because the `breed.id` column contains no value of `10`.

Now, about SQLAlchemy relationships. This is a helpful utility provided by SQLAlchemy that will query a related object for you, so you don't need to refer to the FK all the time. It comes in handy, but you should remember that they're _not_ required. Following the above example, those could be represented by the following SQLA models:
```python
class Breed(Model):
    __tablename__ = 'breed'
    id = Column(Integer, primary_key=True)
    type = Column(Text, nullable=False, unique=True)

    # We are adding this because it's convenient, 
    # not because it's necessary:
    dogs = relationship('Dog', foreign_keys='Dog.breed_id')

    def __repr__(self):
        return '<Breed {}: {}>'.format(self.id, self.type)

class Dog(Model):
    __tablename__ = 'dog'
    id = Column(Integer, primary_key=True)
    name = Column(Text, nullable=False)
    breed_id = Column(Integer, ForeignKey('breed.id'), nullable=False)

    # We are adding this because it's convenient, 
    # not because it's necessary:
    breed = relationship('Breed', foreign_keys='Breed.id')

    def __repr__(self):
        return '<Dog {}: {} is a {}>'.format(self.id, self.name, self.breed.type)
```

If we assume our tables are populated like the earlier example demonstrates, we can do this:
```python
>>> from models import Dog, Breed
>>> a_dog = Dog.query.get(1)
>>> a_dog.id
1
>>> a_dog.name
'Fido'
>>> a_dog.breed_id
2
>>> a_dog.breed.id
2
>>> a_dog.breed.id == a_dog.breed_id  # This will always be True
True
>>> a_dog.breed
'<Breed 2: dalmatian>'
>>> dalmatian = Breed.query.get(2)
>>> dalmatian.id
2
>>> dalmatian is a_dog.breed
True
>>> dalmatian.dogs
['<Dog 1: Fido is a dalmatian>', '<Dog 3: Rover is a dalmatian>']
```
As you can see, the FK column attribute, `Dog.breed_id`, is always an integer corresponding to a particular `Breed.id` value. The `Dog.breed` relationship is merely a shortcut for loading the associated `Breed`.

I'll get to addressing 3 and 4 later, but wanted to send these out first :)